### PR TITLE
Update owin 3.x->4.x

### PR DESCRIPTION
**Pull request by ryuyu**
Updating to accommodate upcoming chrome change for SameSite cookie.

### CODE REVIEW

File: src/NuGetGallery.Services/NuGetGallery.Services.csproj [hunk 1]
```diff
@@ -269,7 +269,7 @@
       <Version>1.0.0</Version>
     </PackageReference>
     <PackageReference Include="Microsoft.Owin">
-      <Version>3.0.1</Version>
+      <Version>4.1.0</Version>
     </PackageReference>
     <PackageReference Include="Microsoft.Owin.Host.SystemWeb">
       <Version>3.0.1</Version>
```
Comment (loic-sharma) [inline src/NuGetGallery.Services/NuGetGallery.Services.csproj]: Should we update this package too? There may be more in this project:

* Microsoft.Owin.Security
* Microsoft.Owin.Security.Cookies
* Microsoft.Owin.Security.MicrosoftAccount
* Microsoft.Owin.Security.OpenIdConnect

File: src/NuGetGallery/NuGetGallery.csproj [hunk 2]
```diff
@@ -2257,7 +2257,7 @@
       <Version>2.2.29</Version>
     </PackageReference>
     <PackageReference Include="Microsoft.Owin">
-      <Version>3.0.1</Version>
```
Comment (loic-sharma) [inline src/NuGetGallery/NuGetGallery.csproj]: Looks like this project also has more OWIN dependencies. Should they be updated too?

* Microsoft.Owin.Host.SystemWeb
* Microsoft.Owin.Security
* Microsoft.Owin.Security.Cookies
* Microsoft.Owin.Security.MicrosoftAccount
* Microsoft.Owin.Security.OpenIdConnect

**Comment by loic-sharma** [comment]
According to [the release notes](https://github.com/aspnet/AspNetKatana/wiki/Roadmap#400-release-february-2018), IdentityModel dependencies were updated to 5.2.0. It looks like we depend on version 3.19.4 (see [this](https://github.com/NuGet/NuGetGallery/pull/7813/files/d6f25d0d01ac7df52357a46e82a81a7dee194b58#diff-8110fe0d3615d286a6788c2f19f51e54L2254)). Will this cause run time issues?

File: src/NuGetGallery/NuGetGallery.csproj (hunk 2 above)
Comment (ryuyu) [reply src/NuGetGallery/NuGetGallery.csproj]: Hmm. Makes the most sense to pick up everything at once i guess. I'll update these and see what happens.

**Comment by ryuyu** [comment]
Tested on dev to be working

File: src/NuGetGallery.Services/Authentication/Providers/AzureActiveDirectory/AzureActiveDirectoryAuthenticator.cs [hunk 3]
```diff
@@ -38,7 +39,7 @@ protected override void AttachToOwinApp(IGalleryConfigurationService config, IAp
                 {
                     RedirectToIdentityProvider = notification =>
                     {
-                        if (notification.ProtocolMessage.RequestType == OpenIdConnectRequestType.LogoutRequest)
+                        if (notification.ProtocolMessage.RequestType == OpenIdConnectRequestType.Logout)
```
Comment (shishirx34) [inline src/NuGetGallery.Services/Authentication/Providers/AzureActiveDirectory/AzureActiveDirectoryAuthenticator.cs]: did you test this with a breakpoint?

File: src/NuGetGallery.Services/Authentication/Providers/AzureActiveDirectoryV2/AzureActiveDirectoryV2Authenticator.cs [hunk 4]
```diff
@@ -99,9 +100,9 @@ protected override void AttachToOwinApp(IGalleryConfigurationService config, IAp
             {
                 RedirectUri = siteRoot + _callbackPath,
                 PostLogoutRedirectUri = siteRoot,
-                Scope = OpenIdConnectScopes.OpenIdProfile + " email",
-                ResponseType = OpenIdConnectResponseTypes.CodeIdToken,
-                TokenValidationParameters = new System.IdentityModel.Tokens.TokenValidationParameters() { ValidateIssuer = false },
+                Scope = OpenIdConnectScope.OpenIdProfile + " email",
+                ResponseType = OpenIdConnectResponseType.CodeIdToken,
```
Comment (shishirx34) [inline src/NuGetGallery.Services/Authentication/Providers/AzureActiveDirectoryV2/AzureActiveDirectoryV2Authenticator.cs]: I would just make sure that the constant strings here haven't changed.

File: src/NuGetGallery/Web.config [hunk 5]
```diff
@@ -556,6 +556,10 @@
   </system.diagnostics>
   <runtime>
     <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+			<dependentAssembly>
+				<assemblyIdentity name="Microsoft.IdentityModel.Clients.ActiveDirectory" publicKeyToken="31BF3856AD364E35" culture="neutral"/>
+				<bindingRedirect oldVersion="0.0.0.0-5.2.6.0" newVersion="5.2.6.0"/>
+			</dependentAssembly>
```
Comment (shishirx34) [inline src/NuGetGallery/Web.config]: nit: indentations

**Pull request merged**